packages/static-hosting/lib/static-hosting.ts

```diff
@@ -221,226 +170,107 @@ export class StaticHosting extends Construct {
+    const s3Origin = new S3Origin(bucket);
+
+    const originRequestPolicy = new OriginRequestPolicy(
+      this,
+      "S3OriginRequestPolicy",
+      {
+        headerBehavior:
+          OriginRequestHeaderBehavior.allowList("x-forwarded-host"),
+      }
+    );
 
-    let originConfigs = new Array<SourceConfiguration>();
+    const originCachePolicy = new CachePolicy(this, "S3OriginCachePolicy", {
+      headerBehavior: CacheHeaderBehavior.allowList("x-forwarded-host"),
+      enableAcceptEncodingBrotli: true,
+      enableAcceptEncodingGzip: true,
+    });
 
-    // Add the backend host as an origin
-    if (props.backendHost) {
-      originConfigs.push({
-        customOriginSource: {
-          domainName: props.backendHost,
-        },
-        behaviors: [], // Behaviors will be added below
-      });
+    const errorResponses: ErrorResponse[] = [
+      {
+        httpStatus: 404,
+        responseHttpStatus: 200,
+        responsePagePath: props.errorResponsePagePath ?? "/index.html",
+        ttl: Duration.seconds(0),
+      },
+    ];
+
+    let responseHeadersPolicy: IResponseHeadersPolicy | undefined;
+
+    if (!disableCSP) {
+      const cspHeader = this.generateCSPString(props.csp, props.explicitCSP);
 
-      // Redirect paths
-      if (props.remapBackendPaths) {
-        for (const path of props.remapBackendPaths) {
-          originConfigs[0].behaviors.push(
-            this.createRemapBehavior(path.from, path.to)
-          );
+      responseHeadersPolicy = new ResponseHeadersPolicy(
+        this,
+        "ResponseHeaders",
+        {
+          securityHeadersBehavior: {
+            contentSecurityPolicy: {
+              contentSecurityPolicy: cspHeader,
+              override: true,
+            },
+          },
         }
-      }
+      );
     }
 
-    // Create default origin
-    originConfigs.push({
-      s3OriginSource: {
-        s3BucketSource: bucket,
-        originAccessIdentity: oai,
-      },
-      // if behaviors have been passed via props use them instead
-      behaviors: props.behaviors
-        ? props.behaviors
-        : [
-            {
-              isDefaultBehavior: true,
-            },
-          ],
-    });
+    const defaultBehavior: Writeable<BehaviorOptions> = {
+      origin: s3Origin,
+      viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+      edgeLambdas: props.defaultBehaviorEdgeLambdas,
+      originRequestPolicy: originRequestPolicy,
+      cachePolicy: originCachePolicy,
+      responseHeadersPolicy: responseHeadersPolicy,
+    };
 
-    // Create behaviors to map static content to bucket
-    if (enableStaticFileRemap) {
-      for (const path of this.staticFiles) {
-        originConfigs[originConfigs.length - 1].behaviors.push(
-          this.createRemapBehavior(`*.${path}`, `*.${path}`)
-        );
-      }
-    }
+    const additionalBehaviors: Record<string, Writeable<BehaviorOptions>> = {};
```

Comment:
How do you add additionalBehaviors? It doesn't look like we are using the props here

Reply:
we're defining `additionalBehaviors` and then taking the `reponseHeaderPolicies` (prop name retanied from v1 for easier migration) and just passing it as is.

if additional behaviours are defined per path, it's just iterated over and added as `additionalBehaviors` to the distributions API.

Reply:
misread your comment, this is now done via merging the `additionalBehaviors` and `props.additionalBehaviors`.